### PR TITLE
Add Bitcoin Core Node role doc

### PR DESCRIPTION
**Pull request by m52go**
As discussed [here](https://github.com/bisq-network/roles/issues/66). I'm not sure about some of the details in the Process section, so @sqrrm if you could take a look at that section in particular, it would help.

Once this is done, we should begin making docs for other roles.

### CODE REVIEW

File: btcnode.adoc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+Bisq contributors run a federation of Bitcoin Core nodes to ensure Bisq connects to nodes with consistent Bitcoin Core implementations.
+
+
+== Introduction
+
+In the past, Bisq connected to arbitrary Bitcoin Core nodes, but the prospect of the SegWit2x hard fork in Fall 2017 made it clear this was not ideal—in the case of a hard fork, it would be impossible for Bisq to tell which fork a connected node was running. The SegWit2x hard fork never happened, but if it did, there was a chance transactions would execute on an undesirable fork, and even worse, a chance that transactions would execute on different forks.
+
+Hence the decision to run a federation of Bitcoin nodes known to run the same Bitcoin Core implementation and configuration—Bisq can expect all bitcoin nodes it connects to enforce consistent block validation rules.
+
+An important additional benefit of this setup is added user privacy: bloom filters in bitcoinj are not implemented very well, so connecting only to nodes run by trusted Bisq contributors avoids monitoring from arbitrary public nodes.
```
Comment (ManfredKarrer) [inline btcnode.adoc]: I would rephrase it so something like that:
bloom filters carry serious privacy flaws when connecting to arbitrary bitcoin nodes as chain analysis companies try to spy on SPV wallets. Connecting only to nodes run by trusted Bisq contributors avoids such monitoring attempts from arbitrary public nodes.

File: btcnode.adoc [hunk 2]
```diff
@@ -0,0 +1,95 @@
+= Bitcoin Core Nodes
+
+Bisq contributors run a federation of Bitcoin Core nodes to ensure Bisq connects to nodes with consistent Bitcoin Core implementations.
```
Comment (ManfredKarrer) [inline btcnode.adoc]: I would add here the bloom filer/privacy reason as that is a very strong reason.

File: btcnode.adoc [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+Contact the Bitcoin Core Maintainer to convey your intention of setting up a Bitcoin Core node for Bisq.
+
+=== Step 2. Set it Up
+
+Set up a Bitcoin Core node, making sure to use the configuration https://github.com/bisq-network/bisq-btcnode[specified here] [where is it, actually?]. Also make sure your node runs in a secure environment [ details? ] and that it allows onion connections so Bisq can connect to it.
+
+=== Step 3. Include Node in Bisq Source
+
+Once your node is ready, send a PR to include it as a Bitcoin Core node for Bisq (they're currently specified https://github.com/bisq-network/bisq/blob/ff4e22d5f537ec8e77bffb3fec116b96fa387c46/core/src/main/java/bisq/core/btc/nodes/BtcNodes.java[here]).
```
Comment (cbeams) [inline btcnode.adoc]: Nit: probably better to make links like this refer to `master` vs. a specific commit, just so when they click the link, they get an up-to-date view on things. The downside of course is that if that class is moved or renamed, the link will not work anymore.

File: index.adoc [hunk 4]
```diff
@@ -53,6 +53,7 @@ Docs without hyperlinks haven't been written yet. If you want to write one, <<co
  * <<manual-dispute-payout#, How to issue a manual dispute payout>>
  * <<exchange/howto/run-seednode#, How to run a seednode>>
  * <<exchange/howto/run-price-relay-node#, How to run a pricenode>>
+ * <<btcnode#, Bitcoin Node Maintainer Role>>
```
Comment (cbeams) [inline index.adoc]: Nit: Probably good to nest this under the 'Roles' item or to create a separate subsection called 'Roles'. Refer to the OmniOutliner doc I sent you earlier, @m52go. Not worth holding this PR up though. Just mentioning for the future.

**Comment by cbeams** [comment]
I just realized this is still WIP, so I'm holding off from merging it until you say so, @m52go.

Also please make sure to link to this doc from the btcnode maintainer and operator role issue descriptions. I may need to do that, actually, as you probably don't have rights to do that.

File: btcnode.adoc (hunk 3 above)
Comment (m52go) [reply btcnode.adoc]: Thanks for noticing...not sure how that happened.

**Comment by m52go** [comment]
@cbeams yes I'd like to get feedback from @sqrrm on the Process section. At the moment, there are gaps there that need to be filled before this can be merged.

File: btcnode.adoc [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+ * Operate bitcoin nodes as detailed below in <<process>>.
+ * Monitor communications on the `#bisq-btcnode` Slack channel.footnote:[See link:roles.html#communication[]]
+ * Write a monthly report on the Bitcoin Node Operator <<role-issue>>.footnote:[See link:roles.html#reporting[]]
+
+==== Rights
+
+ * N/A
+
+
+== Process
```
Comment (sqrrm) [inline btcnode.adoc]: I'm not sure what the process to add more nodes should be, but like many of the other roles a requirement would be to already be in good standing within the Bisq community since it's a trusted role. It's not as sensitive as arbitrator or pricenode/seednode operator but it's also impossible to find out if an operator is breaking privacy in any way and thus breaking one of the main purposes of running these federated nodes.

Comment (m52go) [reply btcnode.adoc]: Thanks, I'll add that.

Just 2 outstanding gaps to be filled: 

1. where the Bitcoin Core configuration is actually specified (the bisq-network/btcnode repo seems empty)
2. any details of the "secure environment" full nodes need to run in.

Both these items are from the `Step 2. Set it Up` section.

**Comment by m52go** [comment]
I've removed WIP—this is now a merge candidate.

Feedback from everyone above has been applied. Thank you.

@cbeams I would like to start implementing your outline to the index page, but I do have some questions first (probably best left for another thread). For now I've nested this doc under the Roles bullet but expect that to change soon.

One thing to note: until this restructuring/reshuffling of the index page is started, roles docs will be situated a bit awkwardly on the index page (these docs aren't limited to describing roles...they cover the whole element of Bisq infrastructure involving that role). And we don't have a 'Production Components' section yet (I could create one, but it would only have 1 element in it right now).

To get around this I've made the index page link for this doc go to the #operator anchor on the page.

Not sure if this is worth spending a whole lot of time discussing if it will be changed soon anyway but do let me know if you'd prefer a different arrangement for now.
